src/imagery/i.eodag/i.eodag.py

```diff
@@ -139,54 +151,91 @@ def get_bb(vector=None):
     }
 
 
+def get_aoi(vector=None):
+    """Get the AOI for querying"""
+    # Handle empty AOI
+    if not vector:
+        return get_bb()
+
+    # TODO: Read Vector WKB
+
+
```

Comment:
with gs.vector_info(options["vector_map_name"]) you can get, among other things, the coordinates of the vector bounding box... 

Otherwise, the lines that you removed above, 117-120, would do the trick, i.e., no need for a new function, if a vector is provided, it is used to set the computational region and get the bb from there...

Reply:
The target from get_aoi(), at least that is what I think @ninsbl meant, is that it can get the precise shapely object of the vector... so we can search with a more specific area instead of the bounding box...